R/module_data_summary.R

```diff
@@ -95,31 +94,47 @@ srv_data_summary <- function(id, teal_data) {
         } else if (is.null(summary_table_out)) {
           "no datasets to show"
         } else {
+          any_unsupported <- any(apply(summary_table_out, 1, function(x) all(is.na(x[-1]))))
+          if (any_unsupported) {
+            shinyjs::show("show_unsupported")
+          } else {
+            shinyjs::hide("show_unsupported")
+          }
+
+          summary_table_out[is.na(summary_table_out)] <- ""
+
+          unsupported_icon <- icon(
+            name = "fas fa-exclamation-triangle",
+            title = "Unsupported dataset",
+            `data-container` = "body",
+            `data-toggle` = "popover",
+            `data-content` = "object not supported by the data_summary module"
+          )
           body_html <- apply(
             summary_table_out,
             1,
             function(x) {
-              tags$tr(
+              is_supported <- !all(x[-1] == "")
+              elem <- tags$tr(
                 tagList(
                   tags$td(
-                    if (all(x[-1] == "")) {
-                      icon(
-                        name = "fas fa-exclamation-triangle",
-                        title = "Unsupported dataset",
-                        `data-container` = "body",
-                        `data-toggle` = "popover",
-                        `data-content` = "object not supported by the data_summary module"
-                      )
-                    },
+                    if (!is_supported) unsupported_icon,
                     x[1]
                   ),
                   lapply(x[-1], tags$td)
                 )
               )
+              if (!is_supported && is_unsupported_hidden()) {
+                shinyjs::hidden(elem)
+              } else {
+                elem
+              }
             }
           )
 
-          header_labels <- names(summary_table())
+          header_labels <- names(summary_table_out)
+          header_labels[header_labels == "dataname"] <- "Data Name"
+          substring(header_labels, 1, 1) <- toupper(substring(header_labels, 1, 1))
```

Comment:
`get_filter_overview` returns lower-case column names now, they need to be "prettified" so that "Data name", "Subjects", "Obs" are displayed

Reply:
`toupper` changes all the letters to capital letters. maybe you need `tools::toTitleCase`

Reply:
This is exactly what I need, thanks!